### DISCUSSION TITLE
SAML, Renew Signature / x509 Certificate

### DISCUSSION THREAD

**Discussion by PentaPaetzold** [kimai/kimai#4326] [On-Premise: Installation & Updates] [answered]
We use Authentik/SAML with Kimai as documented in https://www.kimai.org/documentation/saml-authentik.html , which works really fine!

Our Certificates in Authentik are let's encryted ones, so they renew quite fast and automagically. Authentik is able to refresh them when they have changed.

But after every refresh, the Signature of the Cert has changed - breaking Kimai Login while the Parameter in local.yaml for the x509cert is wrong and needs to be manually changed each time.

I have had no problem with other services using the same Certificate (but Oauth2), so technically there should be ways to make it work. Removing the x509Cert- Value from Config also breaks Login.

Wondering if there is another parameter to supress the need of x509Cert. If not, any solutions to make it work automatically? The Certs could be made public to kimai i needed.

**Comment by kevinpapst** [comment]
Never needed, don't know. 
You can check configs athttps://github.com/SAML-Toolkits/php-saml#settings

What should definitely work: use a script in the "certbot post hook" that will use `awk` to replace the old cert in `local.yaml` and then rebuild Kimai cache.

**Reply by PentaPaetzold** [reply]
On that page, i see:
```
// Usually x509cert and privateKey of the SP are provided by files placed at
// the certs folder. But we can also provide them with the following parameters
        'x509cert' => '',
        'privateKey' => '
```
So, this is what i want. In Docker- Version of Kimai: where is the certs-folder located (need the folder name for bind mount in docker) and how do i get letsencrypted certs to be used by saml-toolkit of kimai?

**Comment by PentaPaetzold** [comment, marked as answer]
I upped the kimai to version 2.4.0 and tested an empty value:
`x509cert: ''`
Which now works fine. Don' exactly know if it were my cookie-settings (cleared them) or the newer version, but now it works as expected.

Closing this, thank you for your help!

**Reply by kevinpapst** [reply]
Closing discussion means they will be hidden, so I am re-opening for the next person.
